theforeman.org/pipelines/lib/rbenv.groovy

```diff
@@ -0,0 +1,30 @@
+def bundleInstall(version, gemfile=null) {
+    command = "bundle install"
+
+    if (gemfile) {
+        command = "${command} --gemfile=${gemfile}"
+    }
+
+    withRuby(version, "bundle config set path ~/.rubygems")
+    withRuby(version, command)
+}
+
+def bundleExec(version, command, gemfile=null) {
+    command = "bundle exec ${command}"
+
+    if (gemfile) {
+        command = "BUNDLE_GEMFILE=${gemfile} bundle exec ${command}"
+    }
+
+    withRuby(version, command)
+}
+
+def withRuby(version, command) {
+    echo command.toString()
+
+    sh """
+        eval "\$(rbenv init -)"
```

Comment:
This does not seem to work to do this per shell rather than doing it for the whole of the node.

Reply:
My theory is that it has to do with interactive vs non-interactive shell modes. https://github.com/rbenv/rbenv/wiki/Unix-shell-initialization has a nice overview, but doesn't explain if you need to pass some options for non-interactive mode.